napari/layers/base/base.py

```diff
@@ -414,10 +416,34 @@ def data(self):
     def data(self, data):
         raise NotImplementedError()
 
+    @property
     @abstractmethod
-    def _get_extent(self):
+    def _extent_data(self):
+        """(2, D) array: Extent of layer in data coordinates."""
         raise NotImplementedError()
 
+    @property
+    def _extent_world(self):
+        """(2, D) array: Range of layer in world coordinates."""
+        return self._transforms['data2world'](self._extent_data)
+
+    @property
+    def shape(self):
+        """tuple: Size of layer in world coordinates (compatibility)."""
+
+        warnings.warn(
+            (
+                f'layer.shape is deprecated and will be removed '
+                f'in napari > 0.3.7'
```

Comment:
```suggestion
                f'layer.shape is deprecated and will be removed '
                f'please use layer._extent_data[1] to get the shape'
                f'in napari > 0.3.7'
```

Should we provide a suggestion on what to do when `layer.shape` is deprecated?

Reply:
Ah, I now see that the suggestion would likely be different for some layers. I'm not sure the best thing to do there. I just know that as a user, I am usually a little miffed when I get told something is being deprecated without being suggested a path forward.

Reply:
@sofroniewn I see your thumbs up but not any suggestion of how you intend to fix this? I'll point out that we shouldn't suggest attributes beginning in `_`.

Reply:
Will put in something else, will have to think about what makes most sense

Reply:
we can wait on deprecation. i've added a comment in code saying to deprecate after full world coords refactor is done